cmd_gen_seed.go

```diff
@@ -0,0 +1,131 @@
+package main
+
+import (
+	"crypto/rand"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/jessevdk/go-flags"
+	"github.com/lightningnetwork/lnd/aezeed"
+)
+
+const (
+	defaultEntropyBytes = 16
+)
+
+type jsonSeed struct {
+	Seed     string `json:"seed"`
+	Birthday int64  `json:"birthday_timestamp"`
+}
+
+type genSeedCommand struct {
+	EntropySourceFile string            `long:"entropy-source-file" description:"The file descriptor to read the seed entropy from; if set lndinit will read exactly 16 bytes from the file, otherwise the default crypto/rand source will be used"`
+	PassphraseFile    string            `long:"passphrase-file" description:"The file to read the seed passphrase from; if not set, no seed passphrase will be used, unless --passhprase-k8s is used"`
+	PassphraseK8s     *k8sSecretOptions `group:"Flags for reading seed passphrase from Kubernetes" namespace:"passphrase-k8s"`
+	Output            string            `long:"output" short:"o" description:"Output format" choice:"raw" choice:"json"`
+}
+
+func newGenSeedCommand() *genSeedCommand {
+	return &genSeedCommand{
+		Output: outputFormatRaw,
+		PassphraseK8s: &k8sSecretOptions{
+			Namespace: defaultK8sNamespace,
+		},
+	}
+}
+
+func (x *genSeedCommand) Register(parser *flags.Parser) error {
+	_, err := parser.AddCommand(
+		"gen-seed",
+		"Generate an lnd wallet seed",
+		"Generate a fresh lnd wallet seed (aezeed) with 16 bytes of "+
+			"entropy read from the given entropy file or the "+
+			"system's default cryptographic entropy source; the "+
+			"seed is printed to stdout, either as raw text or "+
+			"formatted as JSON",
+		x,
+	)
+	return err
+}
+
+func (x *genSeedCommand) Execute(_ []string) error {
+	// First find out if we want to set a seed passphrase.
+	var (
+		passPhrase string
+		err        error
+	)
+	switch {
```

Comment:
Question: should we default to always require a passphrase? I think it makes sense.

Reply:
This will make it hard to use an existing seed that was created without a seed passphrase. So I think we shouldn't. Otherwise users would first need to manually convert their seeds before they could use this tool.

Reply:
SGTM